### ISSUE TITLE
Unknown index entry format 13110000

### ISSUE THREAD

**Issue by jamespharaoh** [gitpython-developers/GitPython#551]
I'm using gitpython to do my own custom stashing, but it seems to be writing out a corrupt index file in the process. Specifically, when I run git commands afterwards, I get the above error, which I can't see referenced anywhere.

Here's the code:

```python
self.stashed_head_tree = (
    self.git_repo.head.commit.tree)

self.stashed_index_tree = (
    self.git_repo.index.write_tree ())

self.stashed_index_commit = (
    git.Commit.create_from_tree (
        self.git_repo,
        self.stashed_index_tree,
        "Stashed index"))

self.stashed_index_tag = (
    git.Tag.create (
        self.git_repo,
        "wbs/stashed-index",
        self.stashed_index_commit,
        force = True))

self.git_repo.git.add (
    "--all",
    ".")

self.stashed_working_tree = (
    self.git_repo.index.write_tree ())

self.stashed_working_commit = (
    git.Commit.create_from_tree (
        self.git_repo,
        self.stashed_working_tree,
        "Stashed working tree"))

self.stashed_working_tag = (
    git.Tag.create (
        self.git_repo,
        "wbs/stashed-working",
        self.stashed_working_commit,
        force = True))

self.git_repo.index.reset (
    working_tree = True)
```

This has started happening after merging in some projects using git subtree, so I imagine that some of the data there is triggering the bug. Specifically, I've merged branch stable-2.1 of the ansible project, here. I notice that there is a file with non-ascii characters in it, and I think this is causing the problem.

Can anyone shed any light on this please? For the time being, I have simply removed the file, and everything is working ok.

**Comment by ankostis** [comment]
Dear @jamespharaoh, would it be possible to provide a shell-script / and then python code introducing the problem with the unicode file?
Something like this:

```
$ git init
$ echo "καλημερα" > f1.txt
$ git add f1.txt && git commit -m 'c1'
```

Then the python code has to be as short as possible to identify the problermatic point.

**Comment by jamespharaoh** [comment]
I'll have a go at this when I have time... Busy busy busy at the moment ;)

**Comment by Byron** [comment]
@jamespharaoh In the meanwhile, you might be able to workaround the issue by using git commands directly (if at all feasible), e.g. using `repo.git`.

**Comment by jamespharaoh** [comment]
Yeah I've worked around it, by renaming the file, but obviously am keen to fix the problem. I'll try and submit a patch myself if possible, but will definitely provide a test case either way. It will be less effort to get it fixed than forget about it and have the same problem in one of my other 100 or so projects that share this code ;)

**Comment by jamespharaoh** [comment]
Actually, I should probably write/release something related to what I'm doing which triggers this. I've basically implemented my own functionality similar to "git stash", but which treats different parts of the tree separately, specifically because i'm managing external dependencies by importing them. This seems like it might be useful to move out into a library in its own right, and building it on top of gitpython (with errors fixed) has been relatively simple. That said, I might propose some additions to the API and/or docs as well, some of it was not too obvious.

**Comment by Byron** [comment]
I am very much looking forward to it!
